subprojects/core/src/main/java/org/gradle/cache/internal/GradleUserHomeCacheCleanupEnablement.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.cache.internal;
+
+import org.gradle.cache.CleanupAction;
+import org.gradle.initialization.GradleUserHomeDirProvider;
+import org.gradle.util.internal.GUtil;
+
+import java.io.File;
+import java.util.Properties;
+
+public class GradleUserHomeCacheCleanupEnablement implements CacheCleanupEnablement {
+    public static final String CACHE_CLEANUP_PROPERTY = "org.gradle.cache.cleanup";
+
+    private final GradleUserHomeDirProvider userHomeDirProvider;
+
+    public GradleUserHomeCacheCleanupEnablement(GradleUserHomeDirProvider userHomeDirProvider) {
+        this.userHomeDirProvider = userHomeDirProvider;
+    }
+
+    private boolean isDisabled() {
+        File gradleUserHomeDirectory = userHomeDirProvider.getGradleUserHomeDirectory();
+        File gradleProperties = new File(gradleUserHomeDirectory, "gradle.properties");
+        if (gradleProperties.isFile()) {
+            Properties properties = GUtil.loadProperties(gradleProperties);
```

Comment:
This property was purposefully excluded from the buildscript/local project properties because it isn't a per-project property. I don't think there's much value in taking it from the command-line either, as the second you run without it, cleanup would occur.

Reply:
Yeah, I think this was intentional and I feel like we shouldn't change it.  We _want_ to couple this to the Gradle user home it applies to.  We don't want two different builds setting this property differently and then it being confusing to the user why things are still getting cleaned up, when they have this property in their project gradle.properties.  If it can _only_ be in GUH, then there is only one source of truth.  I think CLI is debatable, but we eventually want to just deprecate this property in place of configurable values in the build model, so I don't think it's worthwhile investing in it at this point.